packages/cardano-graphql-services/src/DataProjection/blockHandlers/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import { Asset, BigIntMath } from '@cardano-sdk/core';
+import { AssetMetadata } from '../../MetadataClient/types';
+import { Schema, isAlonzoBlock, isMaryBlock } from '@cardano-ogmios/client';
+
+export const getBlockType = (block: Schema.Block) => {
+  let b: Schema.BlockMary | Schema.BlockAlonzo | undefined;
+  if (isAlonzoBlock(block)) {
+    b = block.alonzo as Schema.BlockAlonzo;
+  } else if (isMaryBlock(block)) {
+    b = block.mary as Schema.BlockMary;
+  }
+  return b;
+};
+
+export const mapMetadata = (serverAssetMetadata: AssetMetadata): Asset.TokenMetadata => ({
+  decimals: serverAssetMetadata.decimals.value,
+  desc: serverAssetMetadata.description?.value,
+  icon: serverAssetMetadata.logo?.value,
+  name: serverAssetMetadata.name?.value,
+  ticker: serverAssetMetadata.ticker?.value,
+  url: serverAssetMetadata.url?.value
+  // Missing fields: sizedIcons
+});
+
+const mapAssets = (assets: { [k: string]: Schema.AssetQuantity } | undefined) =>
+  // Missing: transactionOutput, asset
+  assets ? Object.entries(assets).map(([name, quantity]) => ({ asset: { name }, quantity })) : [];
+
+const mapOutput = (output: Schema.TxOut) =>
+  // Missing: index, transaction, datumHash
+  ({
+    address: output.address,
+    value: { assets: mapAssets(output.value.assets), coin: output.value.coins }
+  });
+
+const mapInput = (input: Schema.TxIn) =>
+  // Missing: address:, redeemer, sourceTransaction, transaction, value
```

Comment:
- address and value: it should be usually available in our database, this is data from previous outputs. Should query those during our "query" step. Let's change our schema so that they're nullable (it's probably the correct model anyway, as there might not be any address for 1st transaction?) and leave this out of scope for this PR. Might end up just removing those fields.
- redeemer: let's leave this out of scope of this PR
- sourceTransaction: graph edge to [this](https://github.com/CardanoSolutions/ogmios/blob/master/clients/TypeScript/packages/schema/src/index.ts#L1104)
- transaction: graph edge to current Transaction object